app/controllers/charges_controller.rb

```diff
@@ -5,6 +5,13 @@
 class ChargesController < ApplicationController
   # POST /charges
   def create
+    # Validate this not a duplicate charge
+    idempotency_key = charge_params[:idempotency_key]
+    existing_event = ExistingEvent.new(idempotency_key: idempotency_key, event_type: 'charges_create')
+    unless existing_event.save
+      return json_response(existing_event.errors, :bad_request)
```

Comment:
Could you throw new Exception here instead? And handle it in the ExceptionHandler?

Reply:
Good call!